### DISCUSSION TITLE
Adding a code-based typing challenge, eventually expanding to more languages, etc.

### DISCUSSION THREAD

**Discussion by spaghettic0de** [monkeytypegame/monkeytype#1726] [Ideas]
Hello, 

I'm certain I'm not the first person to think of this, but I think it would be pretty cool if you could add a feature that tracks WPM on code snippets in a user's coding language of choice. I think that personally this would be useful and would probably have the most appeal to people who do competitive programming, but even regular users who also happen to code from time to time could benefit -- primarily I envision it testing the ability to type sequences of letters and symbols that are common in code (such as semicolon, curly brace, square bracket, and more). Additionally, it would be fairly simple to set this up (assuming users don't complain if the feature isn't super robust from the outset). I think that it would require an upper bound of around 200 1-5 line code snippets, and that the user would be tasked with typing the whole snippet (including newline characters) during the 30seconds allotted to track WPM. If this is a feature that also sounds intriguing to you, I'd be happy to make a JSON file with around 200 snippets from C++ that I believe adequately cover the range of different weird syntactical phrases you could encounter. I also think that it could potentially be appealing to users if they can see the meaning behind the code snippets they've typed -- currently, I'm thinking that the feature will look something like this:

`int main () { 
    std::cout << "Hello World!" << std::endl;
    return 0;
}`
You just wrote a program that prints Hello World in c plus plus! 

The idea is that the user will be encountered with this entire screen, so they will be tasked with typing the program as well as the 1 line explanation of the code snippet's function underneath. I think that this would be a fun inclusion -- if you're at all interested in exploring it as well, let me know and I can take out a pull request with my initial thoughts in code. Also, if you have any feedback about why this design would / wouldn't work, please let me know as well!

**Comment by Miodec** [comment]
You should already be able to select a code language and enter quote mode.
